go-controller/pkg/ovn/libovsdbops/router.go

```diff
@@ -0,0 +1,312 @@
+package libovsdbops
+
+import (
+	"fmt"
+	"net"
+
+	libovsdbclient "github.com/ovn-org/libovsdb/client"
+	"github.com/ovn-org/libovsdb/model"
+	libovsdb "github.com/ovn-org/libovsdb/ovsdb"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/nbdb"
+)
+
+func buildRouterNAT(
+	natType nbdb.NATType,
+	externalIP string,
+	logicalIP string,
+	logicalPort string,
+	externalMac string,
+	externalIDs *map[string]string,
+) *nbdb.NAT {
+	nat := &nbdb.NAT{
+		Type:       natType,
+		ExternalIP: externalIP,
+		LogicalIP:  logicalIP,
+		Options:    map[string]string{"stateless": "false"},
+	}
+
+	if logicalPort != "" {
+		nat.LogicalPort = &logicalPort
+	}
+
+	if externalMac != "" {
+		nat.ExternalMAC = &externalMac
+	}
+
+	if externalIDs != nil {
+		nat.ExternalIDs = *externalIDs
```

Comment:
This seems wrong to me. `ExternalIDs` is not a pointer (optional field in the NAT table), see: https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/nbdb/nat.go#L21

Reply:
Oh right, I see now that the argument is a pointer too and this just de-references that. Please change that to a normal value.